gov2/demotools/question_test.go

```diff
@@ -50,6 +50,64 @@ func TestQuestioner_AskFalse(t *testing.T) {
 	}
 }
 
+func AskBoolWithDefault_ByArguments(t *testing.T, expected string, defaultValue string, entry string, success bool) {
+	questioner := Questioner{reader: bufio.NewReader(mockReader{data: []byte(entry + "\n")})}
+	actual := questioner.AskBoolWithDefault("Bool?", expected, defaultValue)
+	if actual != success {
+		t.Errorf("Expected %v, got %v\n", success, actual)
+	}
+}
+
+func TestQuestioner_AskBoolWithDefault_yY_Blank(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "y", "Y", "", true)
+}
+
+func TestQuestioner_AskBoolWithDefault_yN_Blank(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "y", "N", "", false)
+}
+func TestQuestioner_AskBoolWithDefault_nY_Blank(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "n", "Y", "", false)
+}
+func TestQuestioner_AskBoolWithDefault_nN_Blank(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "n", "N", "", true)
+}
+
+func TestQuestioner_AskBoolWithDefault_yY_y(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "y", "Y", "y", true)
+}
+func TestQuestioner_AskBoolWithDefault_yN_y(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "y", "N", "y", true)
+}
+func TestQuestioner_AskBoolWithDefault_nY_y(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "n", "Y", "y", false)
+}
+func TestQuestioner_AskBoolWithDefault_nN_y(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "n", "N", "y", false)
+}
+
+func TestQuestioner_AskBoolWithDefault_yY_n(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "y", "Y", "n", false)
+}
+func TestQuestioner_AskBoolWithDefault_yN_n(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "y", "N", "n", false)
+}
+func TestQuestioner_AskBoolWithDefault_nY_n(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "n", "Y", "n", true)
+}
+func TestQuestioner_AskBoolWithDefault_nN_n(t *testing.T) {
+	AskBoolWithDefault_ByArguments(t, "n", "N", "n", true)
+}
```

Comment:
I think a lot of these are redundant, because y or n doesn't matter, it's more like a or b, and aA or bB are equivalent, so your set of cases should be:
```
aA""
aB""
aAa
aAb
aBa
aBb
```
It might not be fewer lines, but I might have made a struct and then a list of objects, like:
```
type testArgs struct {
    expected string
    defaultVal string
    entry string
    result bool
}

tests := []testArgs{
   {"a", "A", "", true},
   ... etc ...
}